app/src/main/java/com/rahobbs/todo/fragments/TodoFragment.java

```diff
@@ -0,0 +1,228 @@
+package com.rahobbs.todo.fragments;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.design.widget.Snackbar;
+import android.support.v4.app.Fragment;
+import android.support.v4.app.FragmentManager;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.CheckBox;
+import android.widget.CompoundButton;
+import android.widget.EditText;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.firebase.client.Firebase;
+import com.rahobbs.todo.helpers.Feedback;
+import com.rahobbs.todo.R;
+import com.rahobbs.todo.helpers.TodoItem;
+import com.rahobbs.todo.helpers.TodoLab;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Locale;
+import java.util.UUID;
+
+/**
+ * Created by rachael on 2/1/16.
+ * Fragment that contains a single to-do item detailed view
+ */
+public class TodoFragment extends Fragment {
+
+    private static final String ARG_TODO_ID = "todoId";
+    private static final String DIALOG_DATE = "DialogDate";
+    private static final int REQUEST_DATE = 0;
+
+    private TodoItem mTodo;
+    private EditText mTitle;
+    private EditText mDetails;
+    private TextView mDueDateTextField;
+    private CheckBox mCompletedCheckbox;
+    private LinearLayout mDateComponents;
+
+    public static TodoFragment newInstance(UUID todoId) {
+        Bundle args = new Bundle();
+        args.putSerializable(ARG_TODO_ID, todoId);
+        TodoFragment fragment = new TodoFragment();
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        Firebase.setAndroidContext(getContext());
```

Comment:
I don't know anything about `Firebase` but by experience with SDK is that they usually like to be instantiated in the `Application` class. Because the `Application` will be alive throughout the entire state of the app while this activity might not be. 

But I could be totally wrong and have no idea what I'm talking about